src/style-spec/reference/v8.json

```diff
@@ -1025,7 +1025,7 @@
     },
     "symbol-sort-key": {
       "type": "number",
-      "doc": "Sorts features in ascending order based on this value. Features with a higher sort key will appear above features with a lower sort key when they overlap. Features with a lower sort key will have priority over other features when doing placement.",
+      "doc": "Sorts features in ascending order based on this value. Features with a higher sort key will appear above other features when they overlap. Features with a lower sort key will have priority over other features during placement when they do not overlap.",
```

Comment:
I recommend that we make the doc very clear with an example.

For example, when symbol collision detection is enabled, features with a lower symbol-sort-key will be placed before features with a higher symbol-sort key.  When symbol collision detection is disabled, features with a higher symbol-sort-key will be drawn on top of features witha  lower symbol-sort-key if they overlap.

Reply:
Hm, your example is not an example as much as the only way which this distinction can be triggered. “when features overlap” is intended to mean “when you have them overlap with `*-allow-overlap”. That’s the main way to “enable/disable symbol collision detection.” I think leading with “when symbol collision detection is enabled/disabled...” is more confusing when we don’t have a property named `”allow-collision-detection”`, but I’ll add a commit that borrows some of this phrasing.